gateway/reverse_proxy_test.go

```diff
@@ -2023,3 +2024,45 @@ func TestQuotaResponseHeaders(t *testing.T) {
 	})
 
 }
+
+func BenchmarkLargeResponsePayload(b *testing.B) {
+	ts := StartTest(func(_ *config.Config) {})
+	b.Cleanup(func() {
+		ts.Close()
+	})
+
+	// Create a 500 MB payload of zeros
+	payloadSize := 500 * 1024 * 1024 // 500 MB in bytes
+	largePayload := bytes.Repeat([]byte("x"), payloadSize)
+
+	largePayloadHandler := func(w http.ResponseWriter, _ *http.Request) {
+
+		// Write the payload to the response writer
+		w.Header().Set("Content-Type", "application/octet-stream")
+		w.Header().Set("Content-Length", strconv.Itoa(payloadSize))
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write(largePayload)
+	}
```

Comment:
- The fix is to assert that err is nil.
- Don't add leading `\n` after `{` (remove leading whitespace).